src/devices/cpu/upd177x/upd177x.h

```diff
@@ -0,0 +1,128 @@
+// license:BSD-3-Clause
+// copyright-holders:Wilbert Pol
+/*****************************************************************************
+
+    uPD177x
+
+******************************************************************************
+
+
+               ___ ___
+      PB4   1 |*  u   | 28  PB3
+      PB5   2 |       | 27  PB2
+      PB6   3 |       | 26  PB1
+      PB7   4 |       | 25  PA8
+      PB8   5 |   u   | 24  PA7
+      /AC   6 |   P   | 23  PA6
+   /PW ON   7 |   D   | 22  PA5
+      Vdd   8 |   1   | 21  PA4
+       X1   9 |   7   | 20  PA3
+       X0  10 |   7   | 19  PA2
+   SOUND2  11 |   1   | 18  PA1
+      Vda  12 |       | 17  CH2
+   SOUND1  13 |       | 16  /EXINT
+      GND  14 |_______| 15  CH1
+
+****************************************************************************/
+
+
+#ifndef MAME_CPU_UPD177X_UPD177X_H
+#define MAME_CPU_UPD177X_UPD177X_H
+
+class upd177x_cpu_device : public cpu_device
+						, public device_sound_interface
+{
+public:
+	upd177x_cpu_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock);
+
+	// configuration helpers
+	auto pb_out_cb() { return m_pb_out_cb.bind(); }
+
+	void pa_w(u8 data) { m_pa = data; }
+	void pb_w(u8 data) { m_pb = data; }
+
+protected:
+	upd177x_cpu_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, u32 clock);
+
+	// device-level overrides
```

Comment:
Please don’t copy/paste this comment.

Reply:
It’s the “level” comments that are a problem (“device-level overrides”, “interface-level overrides, etc.) – they make no sense.  If you want a comment for this section, write “device_t implementation” or similar.

Reply:
I have removed those comments